src/main/java/com/mathworks/ci/MatlabCommandStepExecution.java

```diff
@@ -43,10 +43,15 @@ public Void run() throws Exception {
         
         workspace.mkdirs();
         
-        int res = execMatlabCommand(workspace, launcher, listener, env);
+        int exitCode = execMatlabCommand(workspace, launcher, listener, env);
 
-        getContext().setResult((res == 0) ? Result.SUCCESS : Result.FAILURE);
-        
+        if(exitCode == 0){
+            getContext().setResult(Result.SUCCESS);
+            return null;
+        }
+
+        // throw an exception if return code is non-zero
+        stop(new MatlabExecutionException(exitCode));
```

Comment:
Does a the result not have to be set here?

Reply:
Nope. This will raise exception to pipeline execution level and automatically set build result to fail if exception is not handled by user in pipeline script.